app/models/request/actions/InboundMessageTransformer.scala

```diff
@@ -67,12 +69,15 @@ class InboundMessageTransformer @Inject()(implicit ec: ExecutionContext) extends
   }
 
   //TODO: Consider moving this into MessageResponse
-  private[models] def messageResponse(code: Option[String]): Option[MessageResponse] = code match {
+  private[models] def messageResponse(code: Option[String], channel: ChannelType): Option[MessageResponse] = code match {
     case Some(MessageType.GoodsReleased.code)             => Some(GoodsReleasedResponse)
     case Some(MessageType.ArrivalRejection.code)          => Some(ArrivalRejectedResponse)
     case Some(MessageType.UnloadingPermission.code)       => Some(UnloadingPermissionResponse)
     case Some(MessageType.UnloadingRemarksRejection.code) => Some(UnloadingRemarksRejectedResponse)
-    case _                                                => None
+    case Some(MessageType.XMLSubmissionNegativeAcknowledgement.code) =>
+      logger.error(s"Received the message ${MessageType.XMLSubmissionNegativeAcknowledgement.code} for the $channel channel")
+      Some(XMLSubmissionNegativeAcknowledgementResponse)
```

Comment:
I appreciate this is logging at an extreme level, but is there any place in the incoming journey that will audit the 917 so we can at least look at an obfuscated 917 to understand the errors?

Reply:
There is no separate auditing for 917 it follows the same structure that we have for all incoming messages

Reply:
We have configured an alert for this message especially

Reply:
we won't be able to diagnose the cause of the problem with a simple alert though. Sure we'll know it has happened, but not WHY. That's the value of an audit.

Reply:
Pagerduty alerts tell us it has happened, not what the core issue is. We need some way of identifying what the actual problem is with the XML. We can’t just look it up in the database in production.

Reply:
I'd overlooked that we already audit incoming events. Issue resolved.